src/compiler/scanner.ts

```diff
@@ -1799,7 +1799,21 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
                 // position 0 because UTF-8 decode will fail and produce U+FFFD.
                 // If that happens, just issue one error and refuse to try to scan further;
                 // this is likely a binary file that cannot be parsed
-                if (ch === CharacterCodes.replacementCharacter) {
+                let isBinary = ch === CharacterCodes.replacementCharacter;
+                // See if this is an MPEG Transport Stream, where every 188th byte is "G" and the rest is garbage.

```

Comment:
Is an MPEG transport stream guaranteed to be a multiple of 188 bytes long?

Reply:
Yes, unless it's malformed. But, I actually think I should dump this PR and instead just check the first 80 or 256 characters for replacementCharacter instead. Curious what people think.

Reply:
It's probably fine, but my concern is really more the case that at any arbitrary position, when `scan` is called, you will look ahead 188 characters.

If you had the byte position and the raw content length, you could avoid doing any of this, but because we're operating on something parsed into UTF-16, I don't think you have that option.

Reply:
This only happens at `pos == 0`.

Reply:
Missed that, thanks!